### PR TITLE
Fix: torch layer losses keyword arguments in rematscope

### PR DESCRIPTION
**Pull request by Abhinavexists**
Fix: #21861

wrapped the layer call in a small function that accepts only positional arguments. That wrapper remembers the original keyword arguments.
passed this wrapper to `torch.utils.checkpoint`, so replay uses positional args but the wrapper restores the missing kwargs before calling the layer.

### CODE REVIEW

File: keras/src/backend/torch/core.py [hunk 1]
```diff
@@ -673,7 +674,13 @@ def remat(f):
...
 
     def wrapped(*args, **kwargs):
-        return torch.utils.checkpoint.checkpoint(f, *args, use_reentrant=False)
+        if not kwargs:
+            return checkpoint(f, *args, use_reentrant=False)
+
+        def positional_wrapper(*pos_args):
+            return f(*pos_args, **kwargs)
+
+        return checkpoint(positional_wrapper, *args, use_reentrant=False)
```
Comment (hertschuh) [inline keras/src/backend/torch/core.py]: Looking at the documentation, it looks like you can just do:

```python
return torch.utils.checkpoint.checkpoint(f, *args, use_reentrant=False, **kwargs)
```

Is that not the case?

My concern with your approach is that I think it statically binds the kwargs so they cannot be tensors.

Comment (Abhinavexists) [reply keras/src/backend/torch/core.py]: @hertschuh 
PyTorch's checkpoint() doesn't support passing kwargs to the checkpointed function as i tried to use `return torch.utils.checkpoint.checkpoint(f, *args, use_reentrant=False, **kwargs)` because it was my initial approach but it looses there kwargs and only accept positional args.

Regarding the static binding concern: you're right that this could be problematic. Let me add a test case with tensor kwargs to verify gradient tracking works correctly.

if that works here ?

Comment (hertschuh) [reply keras/src/backend/torch/core.py]: Hmm... it works for me: https://github.com/keras-team/keras/pull/21883

Comment (Abhinavexists) [reply keras/src/backend/torch/core.py]: okay.. i was having issue with my type checker which i interpreted for the `return torch.utils.checkpoint.checkpoint(f, *args, use_reentrant=False, **kwargs)` not working , which is now working fine

**Review by hertschuh — APPROVED** [review]
